conseil-api/src/main/scala/tech/cryptonomic/conseil/api/routes/platform/data/mantis/MantisDataRoutesCreator.scala

```diff
@@ -0,0 +1,165 @@
+  private val logsRoute: Route =
+    delegateCall(mantisLogsEndpoint)(filter => operations.fetchLogs(filter.toQuery.withLimitCap(maxQueryResultSize)))
+
+  /** V2 Route implementation for receipts endpoint */
+  private val receiptsRoute: Route = delegateCall(mantisReceiptsEndpoint)(
+    filter => operations.fetchReceipts(filter.toQuery.withLimitCap(maxQueryResultSize))
+  )
+
+  /** V2 Route implementation for contracts endpoint */
+  private val contractsRoute: Route = delegateCall(mantisContractsEndpoint)(
+    filter => operations.fetchContracts(filter.toQuery.withLimitCap(maxQueryResultSize))
+  )
+
+  /** V2 Route implementation for tokens endpoint */
+  private val tokensRoute: Route = delegateCall(mantisTokensEndpoint)(
+    filter => operations.fetchTokens(filter.toQuery.withLimitCap(maxQueryResultSize))
+  )
+
+  /** V2 Route implementation for token transfers endpoint */
+  private val tokenTransfersRoute: Route = delegateCall(mantisTokenTransfersEndpoint)(
+    filter => operations.fetchTokenTransfers(filter.toQuery.withLimitCap(maxQueryResultSize))
+  )
+
+  /** V2 Route implementation for accounts endpoint */
+  private val accountsRoute: Route = delegateCall(mantisAccountsEndpoint)(
+    filter => operations.fetchAccounts(filter.toQuery.withLimitCap(maxQueryResultSize))
+  )
+
+  /** V2 Route implementation for accounts by address endpoint */
+  private val accountsByAddressRoute: Route =
+    delegateCall(mantisAccountByAddressEndpoint)(operations.fetchAccountByAddress)
+
+  /**
+    * Helper method, which validates the network and delegates the call for specific endpoint to the given method `f`.
+    * Note that this method works only for Tuple2 input parameters in the `endpoint`.
+    **/
+  private def delegateCall0[B](
+      endpoint: Endpoint[(String, Option[String]), Option[B]]
+  )(f: () => Future[Option[B]]): Route =
+    endpoint.implementedByAsync {
+      case (network, _) => platformNetworkValidation(network)(f())
+    }
+
+  /**
+    * Helper method, which validates the network and delegates the call for specific endpoint to the given method `f`.
+    * Note that this method works for more generic input type in the `endpoint`.
+    **/
+  private def delegateCall[A, B](
+      endpoint: Endpoint[((String, A), Option[String]), Option[B]]
+  )(f: A => Future[Option[B]]): Route =
+    endpoint.implementedByAsync {
+      case ((network, element), _) => platformNetworkValidation(network)(f(element))
+    }
+
+  /** V2 concatenated routes */
+  override val getRoute: Route =
+    concat(
+      blocksHeadRoute,
+      blockByHashRoute,
+      blocksRoute,
+      transactionsRoute,
+      transactionByIdRoute,
+      logsRoute,
+      receiptsRoute,
+      contractsRoute,
+      tokensRoute,
+      tokenTransfersRoute,
+      accountsRoute,
+      accountsByAddressRoute
+    )
+
+  /** Function for validation of the platform and network with flatten */
+  private def platformNetworkValidation[A](network: String)(
```

Comment:
not for this PR, but this two methods `platformNetworkValidation` and `pathValidation` could be generic helpers in a separate file.